File: java/src/main/java/com/danifoldi/dml/type/DmlString.java
package com.danifoldi.dml.type;

import com.danifoldi.dml.DmlSerializer;

public class DmlString extends DmlCommentableValue {
    private String value;

    public DmlString(String value) {
        this.value = value;
    }

    public String value() {
        return this.value;
    }

    public void value(String value) {
        this.value = value;
    }

    @Override
    public String serialize(int indent) {
        return DmlSerializer.serializeComment(comment(), indent) + DmlSerializer.indent("'" + value.replace("'", "\\'") + "'", indent);
    }

    @Override
    public String toString() {
        return "DmlString{" + value + "}";
    }
}

File: javascript/src/types/DmlString.js
import DmlCommentableValue from './DmlCommentableValue.js'
import DmlSerializer from '../DmlSerializer.js'

class DmlString extends DmlCommentableValue {
  #value
  constructor(value) {
    super()
    this.#value = value
  }

  value(value) {
    if (typeof value === 'undefined') {
      return this.#value
    }

    this.#value = value
  }

  serialize(indent) {
    DmlSerializer.serializeComment(this.comment(), indent)
      + DmlSerializer.indent('\'' + this.#value.replace('\'', '\\\'') + '\'', indent)
  }
}

export default DmlString
